opencti-platform/opencti-graphql/tests/01-unit/domain/stixCyberObservableDomain-test.ts

```diff
@@ -0,0 +1,185 @@
+      const expectedKey = 'File_sha256__File_sha512';
+      const expectedValue = '057aa4a06395c384a2a9d29f499b410ac1da6fc2c10aa61908eea3e67a32b873__c1527cd893c124773d811911970c8fe6e857d6df5dc9226bd8a160614c0cd963a4ddea2b94bb7d36021ef9d865d5cea294a82dd49a0bb269f51f6e7a57f79421';
+      const { key, value } = generateKeyValueForIndicator('StixFile', file.hashes['SHA-512'], file);
+      expect(key).toEqual(expectedKey);
+      expect(value).toEqual(expectedValue);
+    });
+
+    it('should generate key value from hashed observable entity type and name', () => {
+      const expectedKey = 'X509-Certificate_name';
+      const expectedValue = 'certificateName';
+      const { key, value } = generateKeyValueForIndicator('X509-Certificate', certificate.name, certificate);
+      expect(key).toEqual(expectedKey);
+      expect(value).toEqual(expectedValue);
+    });
+
+    it('should generate key value from observable with entity type and name', () => {
+      const expectedKey = 'Observable_pid';
+      const expectedValue = 'observableWithPid';
+      const { key, value } = generateKeyValueForIndicator('Observable', observableWithPid.name, observableWithPid);
+      expect(key).toEqual(expectedKey);
+      expect(value).toEqual(expectedValue);
+    });
+
+    it('should generate key value from observable with entity type and subject', () => {
+      const expectedKey = 'Observable_subject';
+      const expectedValue = 'observableSubject';
+      const { key, value } = generateKeyValueForIndicator('Observable', observableWithSubject.name, observableWithSubject);
+      expect(key).toEqual(expectedKey);
+      expect(value).toEqual(expectedValue);
+    });
+
+    it('should generate key value from observable with entity type and body', () => {
+      const expectedKey = 'Observable_body';
+      const expectedValue = 'observableBody';
+      const { key, value } = generateKeyValueForIndicator('Observable', observableWithBody.name, observableWithBody);
+      expect(key).toEqual(expectedKey);
+      expect(value).toEqual(expectedValue);
+    });
+  });
+
+  describe('generateIndicatorFromObservable ', () => {
+    const observableToGenerateIndicator = {
+      entity_type: 'Observable',
+      name: 'ObservableName'
+    };
+    const input = {
+      createdBy: 'creator',
+      objectLabel: [],
+      objectMarking: [],
+      objectOrganization: [],
+      externalReferences: [],
+    };
+
+    it('should not be able to generate an indicator because of missing pattern', async () => {
+      await expect(() => generateIndicatorFromObservable(testContext, ADMIN_USER, input, observableToGenerateIndicator))
+        .rejects.toThrowError('Cannot create indicator - cant generate pattern.');
+    });
+  });
+
+  describe('addStixCyberObservable', () => {
+    it('should not be able to create if the type is not an observable', async () => {
+      const input = { type: 'report' };
```

Comment:
```suggestion
      const input = { type: ENTITY_TYPE_CONTAINER_REPORT };
```

better to use the constant (the value is Report and not report)